docs/docs/contributing.md

```diff
@@ -6,7 +6,21 @@ You can contribute in many ways:
 
 ### Language Implementations
 
-Implementations in additional languages are welcome.
+Each language implementation exists on its own corresponding branch (e.g., Python on the `python` branch, C++ on the `cpp` branch) with the following structure:
+
+- Core algorithm variants under `dstream/` directory including:
+      - `steady_algo`
+      - `stretched_algo`
+      - `tilted_algo`
+      - Optional hybrid variants (e.g., `hybrid_0_steady_1_stretched_2_algo`)
```

Comment:
```suggestion
      - as needed for particular use cases, secondary algorithms and hybrid variants (e.g., `hybrid_0_steady_1_stretched_2_algo`)
```

Reply:
Should also mention that each algorithm comprises coresponding site selection and lookup implementations